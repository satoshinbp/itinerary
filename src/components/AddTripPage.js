import React from 'react';
import { connect } from 'react-redux';
import TripForm from './TripForm';
import { addTrip } from '../actions/trips';

const AddTripPage = (props) => (
  <div className="container">
    <TripForm
      onSubmit={(trip) => {
        props.dispatch(addTrip(trip));
        props.history.push('/');
      }} />
  </div>
);

export default connect()(AddTripPage);