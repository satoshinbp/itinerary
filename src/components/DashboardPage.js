import React from 'react';
import { Link } from 'react-router-dom';
import TripList from './TripList';
import TripListFilters from './TripListFilters';

const DashboardPage = () => (
  <div className="container">
    <div className="page-header">
      <Link className="btn-push" to="/create">Add Trip</Link>
      <TripListFilters />
    </div>
    <TripList />
  </div>
);

export default DashboardPage;