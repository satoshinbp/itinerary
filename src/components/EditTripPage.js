import React from 'react';
import { connect } from 'react-redux';
import TripForm from './TripForm';
import { editTrip, removeTrip } from '../actions/trips';

const EditTripPage = (props) => {
  return (
    <div className="container">
      <TripForm
        trip={props.trip}
        onSubmit={(trip) => {
          props.dispatch(editTrip(props.trip.id, trip));
          props.history.push('/');
        }} />
      <button
        className="btn-push btn-push--secondary"
        onClick={() => {
          props.dispatch(removeTrip({ id: props.trip.id }));
          props.history.push('/');
        }}>Remove Expense</button>
    </div>
  );
};

const mapStateToProps = (state, props) => ({
  trip: state.trips.find((trip) => trip.id === props.match.params.id)
});

export default connect(mapStateToProps)(EditTripPage);
