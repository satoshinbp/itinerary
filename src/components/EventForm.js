import React, { useState } from 'react'
import { SingleDatePicker } from 'react-dates'
import 'react-dates/lib/css/_datepicker.css'
import 'react-dates/initialize'
import DateFnsUtils from '@date-io/date-fns'
import { TimePicker, MuiPickersUtilsProvider} from '@material-ui/pickers'
import moment from 'moment'

export default (props) => {
  const id = props.event ? props.event.id : props.id
  const tripId = props.event ? props.event.tripId : props.tripId
  const [title, setTitle] = useState(props.event ? props.event.title : '')
  const [date, setDate] = useState(props.event ? moment(props.event.date) : moment(props.date))
  const [startTime, setStartTime] = useState(props.event ? moment(props.event.startTime) : moment(props.date).hour(12).minutes(0))
  const [endTime, setEndTime] = useState(props.event ? moment(props.event.endTime) : moment(props.date).hour(12).minutes(0))
  const [location, setLocation] = useState(props.event ? props.event.location : '')
  const [note, setNote] = useState(props.event ? props.event.note : '')
  const [focused, setFocused] = useState(null)
  const [error, setError] = useState('')

  const onStartTimeChange = time => {
    setStartTime(moment(time))
    if (moment(time).isAfter(endTime)) {
      setEndTime(moment(time))
    }
  }
  const onEndTimeChange = time => {
    setEndTime(moment(time))
    if (moment(time).isBefore(startTime)) {
      setStartTime(moment(time))
    }
  }
  const onSubmit = e => {
    e.preventDefault()

    if (!title) {
      setError('Please provide title.')
    } else {
      setError('')
      props.onSubmit({
        id,
        tripId,
        title,
        date: date.valueOf(),
        startTime: startTime.valueOf(),
        endTime: endTime.valueOf(),
        location,
        note
      })
    }
  }

  return (
    <form className="form" onSubmit={onSubmit}>
      {error && <p className="form__error">{error}</p>}
      <input
        type="text"
        placeholder="title"
        autoFocus
        className="text-input"
        value={title}
        onChange={e => setTitle(e.target.value)}
      />
      <div className="date-time-picker">
        <SingleDatePicker
          date={date}
          onDateChange={date => setDate(moment(date).startOf('day'))}
          focused={focused}
          onFocusChange={({ focused }) => setFocused(focused)}
          isOutsideRange={() => false}
          displayFormat="YYYY/MM/DD"
        />
      </div>
      <div className="date-time-picker">
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <TimePicker value={startTime} onChange={onStartTimeChange} label="Start Time" />
        </MuiPickersUtilsProvider>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <TimePicker value={endTime} onChange={onEndTimeChange} label="End Time" />
        </MuiPickersUtilsProvider>
      </div>
      <input
        type="text"
        placeholder="location"
        className="text-input"
        value={location}
        onChange={e => setLocation(e.target.value)}
      />
      <textarea
        placeholder="Add a note for your event (optional)"
        className="textarea"
        value={note}
        onChange={e => setNote(e.target.value)}
      >
      </textarea>
      <button className="btn-push">Save Event</button>
    </form>
  )
}