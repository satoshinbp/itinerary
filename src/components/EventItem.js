import React from 'react'
import { connect } from 'react-redux'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import moment from 'moment'
import { removeEvent } from '../actions/events'

export const EventItem = ({ dispatch, id, title, startTime, endTime, location, setEventId }) => {
  const confirmRemoveEvent = () => {
    const result = confirm('Are you sure you want to delete this event?');

    if (result) {
      dispatch(removeEvent(id))
    }
  }

  return (
    <div className="schedule-list-item">
      <div className=" schedule-list-item__actions">
        <FontAwesomeIcon className="icon" icon="trash-alt" onClick={confirmRemoveEvent} />
      </div>
      <div className="schedule-list-item__time" onClick={() => setEventId(id)}>
        <FontAwesomeIcon className="icon show-for-desktop" icon="clock" />
        {moment(startTime).format('HH:mm')} ~ {moment(endTime).format('HH:mm')}
      </div>
      <div className="schedule-list-item__title" onClick={() => setEventId(id)}>
        {title}
      </div>
      <div className="schedule-list-item__location" onClick={() => setEventId(id)}>
        {location && <FontAwesomeIcon className="icon show-for-desktop" icon="map-marker-alt" />}
        {location}
      </div>
    </div>
  )
}

export default connect()(EventItem);