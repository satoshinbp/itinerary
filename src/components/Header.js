import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogout } from '../actions/auth'

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="container">
      <div className="header__content">
        <div className="header__left">
          <Link className="header__title" to="/dashboard">
            <h1>Itinerary</h1>
          </Link>
          <img className="logo" src="/images/logo.png"></img>
        </div>
        <button className="btn-link" onClick={startLogout}>Logout</button>
      </div>
    </div>
  </header>
)

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)