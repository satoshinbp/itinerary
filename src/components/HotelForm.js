import React, { useState } from 'react'
import { DateRangePicker } from 'react-dates'
import 'react-dates/lib/css/_datepicker.css'
import 'react-dates/initialize'
import DateFnsUtils from '@date-io/date-fns'
import { TimePicker, MuiPickersUtilsProvider } from '@material-ui/pickers'
import moment from 'moment'

export default (props) => {
  const id = props.hotel ? props.hotel.id : props.id
  const tripId = props.hotel ? props.hotel.tripId : props.tripId
  const [name, setName] = useState(props.hotel ? props.hotel.name : '')
  const [checkInDate, setCheckInDate] = useState(props.hotel ? moment(props.hotel.checkInDate) : moment(props.date))
  const [checkOutDate, setCheckOutDate] = useState(props.hotel ? moment(props.hotel.checkOutDate) : moment(props.date).add(1, 'day'))
  const [checkInTime, setCheckInTime] = useState(props.hotel ? moment(props.hotel.checkInTime) : moment("1500", "hmm"))
  const [checkOutTime, setCheckOutTime] = useState(props.hotel ? moment(props.hotel.checkOutTime) : moment("1100", "hmm"))
  const [ETA, setETA] = useState(props.hotel ? props.hotel.ETA : moment("1500", "hmm"))
  const [ETD, setETD] = useState(props.hotel ? props.hotel.ETD : moment("1100", "hmm"))
  const [location, setLocation] = useState(props.hotel ? props.hotel.location : '')
  const [note, setNote] = useState(props.hotel ? props.hotel.note : '')
  const [focusedInput, setFocusedInput] = useState(null)
  const [error, setError] = useState('')

  let numberOfMonths
  if (window.matchMedia("(min-width: 45rem)").matches) {
    numberOfMonths = 2
  } else {
    numberOfMonths = 1
  }
  
  const onDatesChange = ({ startDate, endDate }) => {
    if (startDate) {
      setCheckInDate(moment(startDate).startOf('day'))
    }
    if (endDate) {
      setCheckOutDate(moment(endDate).startOf('day'))
    }
  }
  const onSubmit = e => {
    e.preventDefault()

    if (!name) {
      setError('Please provide hotel name.')
    } else {
      setError('')
      props.onSubmit({ id,
        tripId,
        name,
        checkInDate: checkInDate.valueOf(),
        checkOutDate: checkOutDate.valueOf(),
        checkInTime: checkInTime.valueOf(),
        checkOutTime: checkOutTime.valueOf(),
        ETA: ETA.valueOf(),
        ETD: ETD.valueOf(),
        location,
        note
      })
    }
  }
  
  return (
    <form className="form" onSubmit={onSubmit}>
      {error && <p className="form__error">{error}</p>}
      <input
        type="text"
        placeholder="name"
        autoFocus
        className="text-input"
        value={name}
        onChange={e => setName(e.target.value)}
      />
      <div className="date-time-picker">
        <DateRangePicker
          numberOfMonths={numberOfMonths}
          startDate={checkInDate}
          startDateId="check_in_date_id"
          endDate={checkOutDate}
          endDateId="check_out_date_id"
          onDatesChange={onDatesChange}
          focusedInput={focusedInput}
          onFocusChange={focusedInput => setFocusedInput(focusedInput)}
          isOutsideRange={() => false}
          minimumNights={1}
          displayFormat="YYYY/MM/DD"
        />
      </div>
      <div className="date-time-picker">
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <TimePicker value={checkInTime} onChange={time => setCheckInTime(moment(time))} label="Check-in Time" />
        </MuiPickersUtilsProvider>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <TimePicker value={checkOutTime} onChange={time => setCheckOutTime(moment(time))} label="Check-out Time" />
        </MuiPickersUtilsProvider>
      </div>
      <div className="date-time-picker">
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <TimePicker value={ETA} onChange={time => setETA(moment(time))} label="ETA" />
        </MuiPickersUtilsProvider>
        <MuiPickersUtilsProvider utils={DateFnsUtils}>
          <TimePicker value={ETD} onChange={time => setETD(moment(time))} label="ETD" />
        </MuiPickersUtilsProvider>
      </div>
      <input
        type="text"
        placeholder="location"
        className="text-input"
        value={location}
        onChange={e => setLocation(e.target.value)}
      />
      <textarea
        placeholder="Add a note for your hotel (optional)"
        className="textarea"
        value={note}
        onChange={e => setNote(e.target.value)}
      >
      </textarea>
      <button className="btn-push">Save Hotel</button>
    </form>
  )
}