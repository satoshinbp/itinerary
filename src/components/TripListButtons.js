import React from 'react'
import { connect } from 'react-redux'
import { v4 as uuidv4 } from 'uuid'
import { showUpcomingTrip, showPastTrip, showAllTrip } from '../actions/filters'

const TripListButtons = (props) => {
  const onFilterChange = e => {
    if (e.target.value === 'upcoming') {
      props.showUpcomingTrip()
    } else if (e.target.value === 'past') {
      props.showPastTrip()
    } else if (e.target.value === 'all') {
      props.showAllTrip()
    }
  }

  return (
    <div className="btn-bar">
      <select className="select" value={props.filters.filter} onChange={onFilterChange}>
        <option className="option" value="upcoming">Upcoming Trip</option>
        <option className="option" value="past">Past Trip</option>
        <option className="option" value="all">All Trip</option>
      </select>
      <button className="btn-push" onClick={() => props.setTripId(uuidv4())}>Add Trip</button>
    </div>
  )
}

const mapStateToProps = state => ({ filters: state.filters })

const mapDispatchToProps = dispatch => ({
  showUpcomingTrip: () => dispatch(showUpcomingTrip()),
  showPastTrip: () => dispatch(showPastTrip()),
  showAllTrip: () => dispatch(showAllTrip()),
});

export default connect(mapStateToProps, mapDispatchToProps)(TripListButtons)