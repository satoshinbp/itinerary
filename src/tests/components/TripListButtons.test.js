import React from 'react'
import { shallow } from 'enzyme'
import { TripListButtons } from '../../components/TripListButtons'

let showUpcomingTrip, showPastTrip, showAllTrip, wrapper

describe.each(['upcoming', 'past', 'all'])('initial filter %s', initFilter => {
  beforeEach(() => {
    showUpcomingTrip = jest.fn()
    showPastTrip = jest.fn()
    showAllTrip = jest.fn()
    wrapper = shallow(
      <TripListButtons
        showUpcomingTrip={showUpcomingTrip}
        showPastTrip={showPastTrip}
        showAllTrip={showAllTrip}
        filter={initFilter}
      />
    )
  })

  test('should render TripListButtons correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('filter changed to upcoming', () => {
    wrapper.find('select').simulate('change', {
      target: { value: 'upcoming' }
    })
    expect(showUpcomingTrip).toBeCalled()
  })

  test('filter changed to past', () => {
    wrapper.find('select').simulate('change', {
      target: { value: 'past' }
    })
    expect(showPastTrip).toBeCalled()
  })

  test('filter changed to all', () => {
    wrapper.find('select').simulate('change', {
      target: { value: 'all' }
    })
    expect(showAllTrip).toBeCalled()
  })
})

test('should set TripId when trip add button clicked', () => {
  const setTripId = jest.fn()
  wrapper = shallow(<TripListButtons setTripId={setTripId} />)
  wrapper.find('button').simulate('click')
  expect(setTripId).toBeCalledWith(expect.any(String))
})