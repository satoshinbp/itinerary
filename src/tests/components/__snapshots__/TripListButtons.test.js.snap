// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial filter all should render TripListButtons correctly 1`] = `
<div
  className="btn-bar"
>
  <select
    className="select"
    onChange={[Function]}
    value="all"
  >
    <option
      className="option"
      value="upcoming"
    >
      Upcoming Trip
    </option>
    <option
      className="option"
      value="past"
    >
      Past Trip
    </option>
    <option
      className="option"
      value="all"
    >
      All Trip
    </option>
  </select>
  <button
    className="btn-push"
    onClick={[Function]}
  >
    Add Trip
  </button>
</div>
`;

exports[`initial filter past should render TripListButtons correctly 1`] = `
<div
  className="btn-bar"
>
  <select
    className="select"
    onChange={[Function]}
    value="past"
  >
    <option
      className="option"
      value="upcoming"
    >
      Upcoming Trip
    </option>
    <option
      className="option"
      value="past"
    >
      Past Trip
    </option>
    <option
      className="option"
      value="all"
    >
      All Trip
    </option>
  </select>
  <button
    className="btn-push"
    onClick={[Function]}
  >
    Add Trip
  </button>
</div>
`;

exports[`initial filter upcoming should render TripListButtons correctly 1`] = `
<div
  className="btn-bar"
>
  <select
    className="select"
    onChange={[Function]}
    value="upcoming"
  >
    <option
      className="option"
      value="upcoming"
    >
      Upcoming Trip
    </option>
    <option
      className="option"
      value="past"
    >
      Past Trip
    </option>
    <option
      className="option"
      value="all"
    >
      All Trip
    </option>
  </select>
  <button
    className="btn-push"
    onClick={[Function]}
  >
    Add Trip
  </button>
</div>
`;
